smarts/core/smarts.py

```diff
@@ -596,15 +612,20 @@ def observe_from(self, vehicle_ids):
     def renderer(self):
         if not self._renderer:
             try:
-                from .renderer import Renderer
+                from .renderer import get_renderer
 
-                self._renderer = Renderer(self._sim_id)
-                if self._scenario:
-                    self._renderer.setup(self._scenario)
-                    self._vehicle_index.begin_rendering_vehicles(self._renderer)
+                self._renderer = get_renderer(self._sim_id, self._config)
+            except ImportError as e:
+                self._log.warning("Import failed: " + repr(e))
+                raise e
             except Exception as e:
                 self._log.warning("unable to create Renderer:  " + repr(e))
                 self._renderer = None
+                raise e
```

Comment:
I have elevated these to re-raise assertions so that people do not scratch their head.

A note here that I may not need the `repr()` if I am re-raising.